### PR TITLE
Properly escape quotes with leading spaces

### PR DESCRIPTION
**Pull request by Almighty-Satan**
[contributing]: https://jda.wiki/contributing/contributing/

## Pull Request Etiquette

- [X] I have checked the PRs for upcoming features/bug fixes.
- [X] I have read the [contributing guidelines][contributing].

### Changes

- [X] Internal code
- [ ] Library interface (affecting end-user code) 
- [ ] Documentation
- [ ] Other: \_____ 

Closes Issue: #2358 

## Description

Fixes an issue that caused quotes with leading spaces to not be escaped.

### CODE REVIEW

**Comment by realSurfy** [comment]
I've tested this, it seems like it escapes everything coming after the input string aswell, could you check?

**Comment by Almighty-Satan** [comment]
> I've tested this, it seems like it escapes everything coming after the input string aswell, could you check?

I don't really understand what you mean by that. Could you please elaborate a bit?

**Comment by realSurfy** [comment]
Basically, what I am doing is:

Convert the markdown from Telegram for an automatic Discord message (announcement)

I am doing so by escaping the text and then convert the Telegram entities into Discord markdown.

ORIGINAL MESSAGE: 
image

ESCAPING PRESENT DISCORD MARKDOWN ON THAT TEXT:
image

DEBUG ENTITIES:
image

Entitites get converted into discord markdown

OUTPUT STRING:
image


As you can see, in the output string everything was not being converted

**Comment by Almighty-Satan** [comment]
> After some more debugging I found out that the problem is that the escape actually replaces one space char instead of just adding the \ right before the quote, as shown:
> 
> ORIGINAL: image
> 
> ESCAPED: image
> 
> This is what ends up breaking my converter, the chars basically just don't add up.

I think that should be fixed now

**Comment by realSurfy** [comment]
Tested, I can confirm it does in fact work properly now

DEBUG MESSAGE AFTER ESCAPING + CONVERSION:
image

**Comment by DV8FromTheWorld** [comment]
This needs tests.

File: src/test/java/MarkdownTest.java [hunk 1]
```diff
@@ -459,5 +462,6 @@ public void testQuote()
         Assertions.assertEquals("\\> Hello World", MarkdownSanitizer.escape("\\> Hello World", true));
         Assertions.assertEquals("\\>\\>\\> Hello World", MarkdownSanitizer.escape("\\>\\>\\> Hello World", true));
         Assertions.assertEquals("Hello > World", MarkdownSanitizer.escape("Hello > World"));
+        Assertions.assertEquals("Hello\n \\> World", MarkdownSanitizer.escape("Hello\n > World"));
```
Comment (MinnDevelopment) [inline src/test/java/MarkdownTest.java:465]:
```suggestion
        Assertions.assertEquals("Hello\n \\> World", MarkdownSanitizer.escape("Hello\n > World"));
        Assertions.assertEquals("Hello\n\\> World", MarkdownSanitizer.escape("Hello\n> World"));
```

**Pull request merged**